collector/nginx_plus.go

```diff
@@ -95,50 +138,93 @@ func (c *NginxPlusCollector) DeleteServerZoneLabels(zoneNames []string) {
+}
+
+// DeleteStreamServerZoneLabels deletes the Stream Server Zone Labels
+func (c *NginxPlusCollector) DeleteStreamServerZoneLabels(zoneNames []string) {
+	c.variableLabelsMutex.Lock()
+	for _, k := range zoneNames {
+		delete(c.streamServerZoneLabels, k)
+	}
+	c.variableLabelsMutex.Unlock()
+}
+
 func (c *NginxPlusCollector) getUpstreamServerLabelValues(upstreamName string) []string {
 	c.variableLabelsMutex.RLock()
 	defer c.variableLabelsMutex.RUnlock()
 	return c.upstreamServerLabels[upstreamName]
 }
 
+func (c *NginxPlusCollector) getStreamUpstreamServerLabelValues(upstreamName string) []string {
+	c.variableLabelsMutex.RLock()
+	defer c.variableLabelsMutex.RUnlock()
+	return c.streamUpstreamServerLabels[upstreamName]
+}
+
 func (c *NginxPlusCollector) getServerZoneLabelValues(zoneName string) []string {
 	c.variableLabelsMutex.RLock()
 	defer c.variableLabelsMutex.RUnlock()
 	return c.serverZoneLabels[zoneName]
 }
 
+func (c *NginxPlusCollector) getStreamServerZoneLabelValues(zoneName string) []string {
+	c.variableLabelsMutex.RLock()
+	defer c.variableLabelsMutex.RUnlock()
+	return c.streamServerZoneLabels[zoneName]
+}
+
 func (c *NginxPlusCollector) getUpstreamServerPeerLabelValues(peer string) []string {
 	c.variableLabelsMutex.RLock()
 	defer c.variableLabelsMutex.RUnlock()
 	return c.upstreamServerPeerLabels[peer]
 }
 
+func (c *NginxPlusCollector) getStreamUpstreamServerPeerLabelValues(peer string) []string {
+	c.variableLabelsMutex.RLock()
+	defer c.variableLabelsMutex.RUnlock()
+	return c.streamUpstreamServerPeerLabels[peer]
+}
+
 // VariableLabelNames holds all the variable label names for the different metrics
 type VariableLabelNames struct {
-	UpstreamServerVariableLabelNames     []string
-	ServerZoneVariableLabelNames         []string
-	UpstreamServerPeerVariableLabelNames []string
+	UpstreamServerVariableLabelNames           []string
+	ServerZoneVariableLabelNames               []string
+	StreamServerZoneVariableLabelNames         []string
+	UpstreamServerPeerVariableLabelNames       []string
+	StreamUpstreamServerPeerVariableLabelNames []string
+	StreamUpstreamServerVariableLabelNames     []string
 }
 
 // NewVariableLabels creates a new struct for VariableNames for the collector
 func NewVariableLabelNames(upstreamServerVariableLabelNames []string, serverZoneVariableLabelNames []string,
 	upstreamServerPeerVariableLabelNames []string) VariableLabelNames {
 	return VariableLabelNames{
-		UpstreamServerVariableLabelNames:     upstreamServerVariableLabelNames,
-		ServerZoneVariableLabelNames:         serverZoneVariableLabelNames,
-		UpstreamServerPeerVariableLabelNames: upstreamServerPeerVariableLabelNames,
+		UpstreamServerVariableLabelNames:       upstreamServerVariableLabelNames,
+		ServerZoneVariableLabelNames:           serverZoneVariableLabelNames,
+		UpstreamServerPeerVariableLabelNames:   upstreamServerPeerVariableLabelNames,
+		StreamUpstreamServerVariableLabelNames: upstreamServerVariableLabelNames,
```

Comment:
Do we need to add an extra arg to this func to have separate stream and http upstream server label names?
```suggestion
		StreamUpstreamServerVariableLabelNames: streamUpstreamServerVariableLabelNames,
```

Reply:
they've the same values at the moment, so instead of adding another arg for the function just use the ones that are already there, no?

Reply:
Right:
* resourceName
* resourceNamespace
* resourceType
* serviceName
* podName

I can't foresee us adding anything stream specific any time soon